pkg/db/postgres.go

```diff
@@ -89,9 +89,13 @@ func (pc *postgresConnection) Exec(query string, args ...interface{}) (sql.Resul
 	return result, err
 }
 
-func (pc *postgresConnection) Begin() (*sql.Tx, error) {
+func (pc *postgresConnection) Begin() (*TxConnection, error) {
 	pc.logger.Debug("Postgres Begin()")
-	return pc.db.BeginTx(context.Background(), &sql.TxOptions{Isolation: sql.LevelRepeatableRead})
+	tx, err := pc.db.BeginTx(context.Background(), &sql.TxOptions{Isolation: sql.LevelRepeatableRead})
+	if err != nil {
+		return nil, err
+	}
+	return &TxConnection{tx: tx, conn: pc, counter: 1, logger: pc.logger}, nil
```

Comment:
why `counter:1` and not using 0 as default? when checking the TX implementation, is the counte always 1 and commits are never happening.

Reply:
Since here the Begin does not call the TxConnection `Begin` function and thus, the counter is not increased by one at this point. But since we have the first initial begin here, the counter needs to start at `1`.

We also thought of refactoring it in a way at which we dont need to do this, but this would lead to a bigger refactoring, and I would like to do it in a follow-up, only if really needed.